### ISSUE TITLE
Why CLIP loss is used in second stage training not in first stage ?

### ISSUE THREAD

**Issue by thuangb** [HFAiLab/clip-gen#2]

**Comment by KinglittleQ** [comment]
The first stage is just to train a VQGAN model. It does not need a CLIP loss.

For more details, please refer to the original paper: CLIP-GEN: Language-Free Training of a Text-to-Image Generator with CLIP

**Comment by thuangb** [comment]
> The first stage is just to train a VQGAN model. It does not need a CLIP loss.
> 
> For more details, please refer to the original paper: CLIP-GEN: Language-Free Training of a Text-to-Image Generator with CLIP

I already read the paper. Please correct me if what I going to say is really stupid but I think that CLIP loss in stage2 is quite useless because from the figure of stage2 training phase, the CLIP loss only included the decoder, the generated image and the CLIP embedding of input image, but they are all frozen so nothing can be updated.

![image](https://user-images.githubusercontent.com/87744278/168517789-e5a84945-b416-4d44-a5b8-90ced0e401f5.png)

**Comment by KinglittleQ** [comment]
https://github.com/HFAiLab/clip-gen/blob/14a313d70cdd222df448d212955e6ba359f88cd0/train_gpt.py#L71-L89

We use a trick as above to backpropagate the CLIP loss gradient to the conditional transformer.

However, we didn't observe improvement with CLIP loss. So we set loss_clip_weight to 0 to reduce the computation overhead:

https://github.com/HFAiLab/clip-gen/blob/14a313d70cdd222df448d212955e6ba359f88cd0/train_gpt.py#L43

**Comment by thuangb** [comment]
> https://github.com/HFAiLab/clip-gen/blob/14a313d70cdd222df448d212955e6ba359f88cd0/train_gpt.py#L71-L89
> 
> We use a trick as above to backpropagate the CLIP loss gradient to the conditional transformer.
> 
> However, we didn't observe improvement with CLIP loss. So we set loss_clip_weight to 0 to reduce the computation overhead:
> 
> https://github.com/HFAiLab/clip-gen/blob/14a313d70cdd222df448d212955e6ba359f88cd0/train_gpt.py#L43

Then why bother include in the paper ? It is extremely misleading, isnt it? Anyway, I'm closing this issue now because you have clear my question

**Comment by KinglittleQ** [comment]
Thank you for your advice.

However, we are not the original authors of this paper. We emailed the author and confirmed some implementation details.

**Comment by thuangb** [comment]
Oh sorry then, I thought this one is official
